hooks/basic/collector.py

```diff
@@ -214,12 +216,84 @@ def on_context_changed(self, settings, item):
         work_path_template = item_info["work_path_template"]
 
         # If defined, add the work_path_template to the item's properties
+        # set/evaluate the correct work_path_template
+        # First check if a work_path_template has been defined
+        if work_path_template:
+            work_tmpl = self.parent.get_template_by_name(work_path_template)
+
+            # template is invalid.
+            if not work_tmpl:
+                # let's try to check if this path fits into any known template
+                work_tmpl = self.sgtk.template_from_path(path)
+
+                if not work_tmpl:
+                    # this template was not found in the template config!
+                    raise TankError("The template '%s' does not exist!" % work_path_template)
+
+                # update the field with correct value so that we can use it everytime for this item
+                work_path_template = work_tmpl.name
```

Comment:
1. The user would know in the GUI what template was used to resolve the path, rather than the one stored in settings (which could be incorrect).
2. We can calculate this at the time of object creation and context change, so that we don't have to repeat this code again.

Reply:
Ah, right. Yeah, I get it now. I think I just read the code through too quickly last time. I thought this was part of the previous block where we already had work_path_template. Moving along...nothing to see here :$